common/zigbeeintegrationplugin.cpp

```diff
@@ -46,9 +48,10 @@ ZigbeeIntegrationPlugin::~ZigbeeIntegrationPlugin()
 void ZigbeeIntegrationPlugin::handleRemoveNode(ZigbeeNode *node, const QUuid &networkUuid)
 {
     Q_UNUSED(networkUuid)
-    Thing *thing = m_thingNodes.key(node);
-    if (thing) {
+    foreach (Thing *thing, m_thingNodes.keys(node)) {
```

Comment:
There might be more than one thing per node in the hash? How could that happen?

Reply:
Power strips, for example.